File: Haskell/day4a.hs
module Day4a where 
import Data.List.Split(splitOn)

intInput :: IO [Int]
intInput = map (parseInt . splitThis) . lines <$> getContents

splitThis :: String -> [[String]]
splitThis s = map (splitOn "-")  (splitOn "," s)

parseInt :: [[String]] -> Int
parseInt [[b1,e1],[b2,e2]]
  | bn1 == bn2  = 1
  | en1 == en2  = 1
  | (bn1 > bn2) && (en1 < en2) = 1
  | (bn1 < bn2) && (en1 > en2) = 1
  | otherwise = 0
  where
    bn1 = read b1 :: Int
    en1 = read e1 :: Int
    bn2 = read b2 :: Int
    en2 = read e2 :: Int

main = do
  pointList <- intInput
  print (sum pointList)

File: src/Day4a.java
import java.util.Scanner;

public class Day4a {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int sum = 0;
        int[] a = new int[2];
        int[] b = new int[2];

        while(sc.hasNext()){
            String line = sc.nextLine();
            String[] strings = line.split("-|,");
            a[0] = Integer.parseInt(strings[0]);
            a[1] = Integer.parseInt(strings[1]);
            b[0] = Integer.parseInt(strings[2]);
            b[1] = Integer.parseInt(strings[3]);
            if(!(((a[0] < b[0]) && (a[1] < b[1])) || ((a[0] > b[0]) && (a[1] > b[1])))){
                sum++;
            }
        }
        System.out.println(sum);
    }
}
